File: VillagesTown/Data/Map/Tile.swift
//
//  Tile.swift
//  VillagesTown
//
//  Created by Claude Code
//

import Foundation
import SwiftUI

struct Tile {
    let coordinates: CGPoint
    var terrain: Terrain
    var strategicResource: Resource?
    var explored: Bool = false
    var owner: String? // Village or faction name

    var isEmpty: Bool {
        return owner == nil
    }

    var movementCost: Int {
        return terrain.movementCost
    }

    var defenseBonus: Double {
        return terrain.defenseBonus
    }
}

File: villages_town_flutter/lib/data/models/tile.dart
import 'dart:ui';
import 'resource.dart';
import 'terrain.dart';

class Tile {
  final Offset coordinates;
  final Terrain terrain;
  final Resource? strategicResource;
  bool explored;
  String? owner;

  Tile({
    required this.coordinates,
    required this.terrain,
    this.strategicResource,
    this.explored = false,
    this.owner,
  });

  bool get isEmpty => owner == null && strategicResource == null;
  int get movementCost => terrain.movementCost;
  double get defenseBonus => terrain.defenseBonus;

  Tile copyWith({
    Offset? coordinates,
    Terrain? terrain,
    Resource? strategicResource,
    bool? explored,
    String? owner,
  }) {
    return Tile(
      coordinates: coordinates ?? this.coordinates,
      terrain: terrain ?? this.terrain,
      strategicResource: strategicResource ?? this.strategicResource,
      explored: explored ?? this.explored,
      owner: owner ?? this.owner,
    );
  }
}
